src/wobjectdefs.h

```diff
@@ -199,41 +215,213 @@ constexpr int summed = sums(Args...);
-
-/** Make a StaticStringList out of many char array  */
-constexpr StaticStringList<> makeStaticStringList() { return {}; }
-template<typename... T>
-constexpr StaticStringList<> makeStaticStringList(StaticStringArray<1> &, T...)
-{ return {}; }
-template<std::size_t N, typename... T>
-constexpr auto makeStaticStringList(StaticStringArray<N> &h, T&...t)
-{ return binary::tree_prepend(makeStaticStringList(t...), StaticString<N>(h)); }
-template<typename... T>
-constexpr StaticStringList<> makeStaticStringList(StaticString<1>, T...)
-{ return {}; }
-template<std::size_t N, typename... T>
-constexpr auto makeStaticStringList(StaticString<N> h, T...t)
-{ return binary::tree_prepend(makeStaticStringList(t...), h); }
-
-
-/** Add a string in a StaticStringList */
-template<std::size_t L, typename T>
-constexpr auto addString(const StaticStringList<T> &l, const StaticString<L> & s) {
-    return binary::tree_append(l, s);
+template<size_t... Vs> IndexArray(IndexPack<Vs...>) -> IndexArray<sizeof...(Vs)>;
+
+
+template<size_t... Sizes> constexpr auto toOffsetArray() {
+    auto offsets = IndexArray<sizeof...(Sizes)>{};
+    auto o = size_t{};
+    auto p = offsets.data;
+    ((*p++ = o, o += Sizes),...);
+    return offsets;
+}
+template<size_t I, size_t... Sizes> constexpr auto getOffset() -> size_t {
+    auto i = 0u;
+    auto o = size_t{};
+    ((i++ < I ? o += Sizes : 0u), ...);
+    return o;
 }
 
+template<size_t... Ns>
+struct StaticStrings {
+    static_assert (((Ns > 0) && ...));
+    static constexpr auto size = sizeof...(Ns);
+    static constexpr auto N = summed<Ns ...>;
+
+    char data[(N > 0 ? N : 1)] = {0};
+
+    constexpr StaticStrings() = default;
+    template<size_t... Ls, class = std::enable_if_t<(sizeof... (Ls) > 0)>>
+    constexpr StaticStrings(StaticString<Ls> ... os) {
+        static_assert (summed<Ls...> == N);
+        auto p = data;
+        auto l = [&](const auto& s) { for (auto c : s.data) *p++ = c; };
+        (l(os), ...);
+    }
+
+    static constexpr auto size_sequence = index_sequence<Ns...>{};
+
+    template<size_t I> constexpr auto operator[] (Index<I>) const {
+        constexpr auto o = getOffset<I, Ns...>();
+        constexpr auto n = (RawArray<size_t, sizeof... (Ns)>{Ns...})[I];
+        auto r = StaticString<n>{};
+        for (auto i = size_t{}; i < n; i++) r.data[i] = data[o + i];
+        return r;
+    }
+};
+template<size_t N, size_t... Ns>
+StaticStrings(StaticString<N>, StaticString<Ns>...) -> StaticStrings<N, Ns...>;
+
+static_assert(StaticStrings(StaticString{"Hello"}, StaticString{"World"}).data[4] == 'o');
+static_assert(
+    (StaticStrings{StaticString{"Hello"}, StaticString{"World"}}[index<1>])
+        .data[0] == 'W');
+
+template<size_t I, size_t...Ns>
+constexpr auto stringFetch(const StaticStrings<Ns...>& s, Index<I>) {
+    if constexpr (I < sizeof... (Ns)) {
+        return s[index<I>];
+    }
+    else {
+        (void)s;
```

Comment:
`if constexpr` basically compiles only the active branch. So variables have to be used in both, otherwise you get warnings.
If you do the same with specialization you simply would not name the argument.